site/_posts/2024-12-25-rustport.md

```diff
@@ -0,0 +1,316 @@
+---
+layout: post
+title: "Fish 4.0: The Fish Of Theseus"
+date: 2024-12-25
+categories: technical
+---
+
+About two years ago, our head maintainer @ridiculousfish opened what quickly became our most-read pull request:
+
+- [#9512 - Rewrite it in Rust](https://github.com/fish-shell/fish-shell/pull/9512)
+
+Truth be told, we did not quite expect that to be as popular as it was.
+It was written as a bit of an in-joke for the fish developers first, and not really as a press release to be shared far and wide.
+We didn't post it anywhere, but it was commented far and wide.
```

Comment:
This sounds like humblebrag and doesn't seem super honest given that "attract (new) contributors" is the primary goal.

Reply:
Attracting them with the *code*, sure. But with the PR? Did you expect that sort of reaction?

(plus the confusion at some of the tone)

Reply:
And I think it's fair to be a bit proud, if you don't let it go to your head.

Like, yeah, it *was* pretty cool to see how many were interested in that?

And we did, as far as I know, not post it anywhere ourselves.

Reply:
Yeah I'm also fine with this as written.

Reply:
Bragging *is* appropriate in a technical blog post, that's not my point.